### PR TITLE
fixed #137: error returned from Capture interface does not contain position and is not wrapped

### LINKED ISSUE #137: wrapped error returned from Capture interface does not contain position and is not wrapped
Sometimes it is useful to return an error from the `Capture(values []string) error` method. However the final returned parser error is not decorated with the position properly.

Actually the returned error is a v2.parseError but it does not contain a valid lexer.Position. Also the original error is discarded and just the message is kept instead of using the wrappingParseError type. 

It seems like that the decorate function in setField (nodes.go:553) should do the job, but it does not use the tokens position. Also it just uses the parseError type instead of the wrappingParseError.

### PR DESCRIPTION
**Pull request by torbenschinke**

### CODE REVIEW

File: nodes.go [hunk 1]
```diff
@@ -38,15 +38,25 @@ type node interface {
...
+	}
+
+	return &tokens[0].Pos
+}
+
+func decorate(err *error, pos *lexer.Position, name func() string) {
 	if *err == nil {
 		return
 	}
-	if perr, ok := (*err).(Error); ok {
```
Comment (alecthomas) [inline nodes.go:45]: It looks like this should
have already been preserving positional information to me, and that all that would need to be done is to ensure the error at the original call site had positional information.

**Review by alecthomas — CHANGES_REQUESTED** [review]
Can you make the commit message descriptive, and this will need some tests to verify the fix.

File: nodes.go (hunk 1 above)
Comment (torbenschinke) [reply nodes.go:45]: yes, it looks like this, but an arbitrary error returned from Capture is not a `participle.Error` so that code is never triggered.

Comment (alecthomas) [reply nodes.go:45]: Similarly too with the `Parseable` interface.

**Comment by alecthomas** [comment]
No response.

**Pull request closed without merging**